From: SeongJae Park <sj@kernel.org>
Subject: mm/damon/stat: return error if monitoring target region is invalid
Date: Mon, 16 Mar 2026 22:11:22 -0700

On 32bit systems with LPAE, the end address might be overflowed or
truncated, resulting in invalid range (the end address is equal or smaller
than the start address).  Return an error for the case.

Link: https://lkml.kernel.org/r/20260317053631.87907-1-sj@kernel.org
Signed-off-by: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/stat.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/damon/stat.c~mm-damon-stat-monitor-all-system-ram-resources-fix
+++ a/mm/damon/stat.c
@@ -188,6 +188,8 @@ static int damon_stat_set_monitoring_reg
 			arg.res.start, addr_unit);
 	addr_range.end = damon_stat_res_to_core_addr(
 			arg.res.end + 1, addr_unit);
+	if (addr_range.end <= addr_range.start)
+		return -EINVAL;
 	return damon_set_regions(t, &addr_range, 1, min_region_sz);
 }
 
_
